### ISSUE TITLE
Installer's Python detection has a CWD-dependent false positive (PEP 420 namespace package import)

### ISSUE THREAD

**Issue by nskidan** [Dicklesworthstone/mcp_agent_mail_rust#128]
## Summary

The bash installer's `detect_python_binary` probe in `install.sh` (around line 917) runs:

```bash
python3 -c "import mcp_agent_mail"
```

When the installer is invoked with CWD equal to `$HOME`, this probe **incorrectly succeeds** even when no Python `mcp_agent_mail` package is installed. The root cause is two-fold:

1. Python implicitly prepends CWD to `sys.path` for `-c` invocations.
2. The Rust install directory is `$HOME/mcp_agent_mail/` by default (per `--dest`), and PEP 420 lets *any* directory matching a module name be imported as a namespace package — no `__init__.py` or `.py` files required.

So the probe ends up importing the Rust install directory itself as a namespace package and concludes Python is present.

## Reproduction

On a system where the Rust install is current and no Python package exists:

```
$ ls ~/mcp_agent_mail/
am  mcp-agent-mail

$ cd ~ && python3 -c "import mcp_agent_mail; print(mcp_agent_mail.__path__)"
_NamespacePath(['/home/USER/mcp_agent_mail'])

$ cd /tmp && python3 -c "import mcp_agent_mail"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'mcp_agent_mail'
```

Verbose log output from a real nightly run (note `binary=1` despite no actual Python install):

```
detect_python_binary:skip_rust_symlink path=/home/ubuntu/.local/bin/am target=/home/ubuntu/mcp_agent_mail/am
detect_python_binary:found importable=python3 -m mcp_agent_mail
detect_python_installation:done clone_found=0 clone=<none> venv=<none> pid=<none>
detect_python:done detected=1 alias=0 binary=1 clone=0 pid=<none>
```

## Impact

The installer concludes "legacy Python installation is still present and takeover/displacement has not been re-run" (per `EXISTING_INSTALL_REPAIR_REASON` at install.sh:4791), enters the repair/reinstall branch, redownloads the binary, and attempts sigstore signature verification. On v0.2.51 that verification then fails with:

```
Error: bundle does not contain cert for verification, please provide public key
main.go:74: error during command execution: bundle does not contain cert for verification, please provide public key
ERR Signature verification failed
```

Net effect: `acfs-update`'s nightly run exits non-zero on a system that needs no actual remediation (the Rust binary is already at the latest version). Reproduced across multiple nights on our VPS.

## Suggested fix (Python detection)

Replace the probe with one that isolates from CWD and rejects namespace packages without a real origin:

```bash
python3 -I -c "import importlib.util, sys; \
  s = importlib.util.find_spec('mcp_agent_mail'); \
  sys.exit(0 if (s and s.origin) else 1)"
```

`-I` strips CWD and `PYTHONPATH` from `sys.path`. Checking `spec.origin` rejects namespace packages (`origin` is `None` for them).

## Note on the sigstore bundle issue

The `bundle does not contain cert for verification` failure on v0.2.51 appears to be a separate concern, and it's only encountered here because of the false-positive detection. If that's a known issue or worth a separate ticket, happy to file one.

## Local workaround for affected users

```
touch ~/.config/mcp-agent-mail/.python-migration-complete
```

This satisfies the `PYTHON_MIGRATION_MARKER` check in `existing_install_can_skip` (install.sh:4770-4795) so the installer treats the system as already-migrated without exercising the false-positive detection. We are running this locally as an interim while waiting for an upstream fix.

**Comment by Dicklesworthstone** [comment]
Fixed in [`c879bea3`](https://github.com/Dicklesworthstone/mcp_agent_mail_rust/commit/c879bea31ad3673cc2593c36b996733fa8c99635).

The root cause is exactly as you diagnosed: `python3 -c "import mcp_agent_mail"` with CWD on `sys.path`, and PEP 420 picking up `$HOME/mcp_agent_mail/` (the Rust install dir) as a namespace package.

Hardened the probe on two axes — both needed, because each defeats a different attack on the import resolver:

1. **Run the probe from `/`**, so CWD-on-sys.path can't pick up the Rust install dir by name collision.
2. **Require `__file__`** — only real (initialized) packages set it; PEP 420 namespace packages do not. This guards against `PYTHONPATH` set by an outer caller (your `cd /tmp` fix alone wouldn't handle that case).

Empirical verification with a fake `$HOME/mcp_agent_mail/` shim and `PYTHONNOUSERSITE=1` (to mask the real installed package on the test host):

- Old probe from $HOME: false-positives ✓ reproduces the bug
- New probe from `/`: correctly fails ✓
- New probe with `PYTHONPATH=$HOME/fake`: also correctly fails ✓ (`__file__` guard catches it)

Closing — please reopen if you hit any false positive on a real takeover/displacement scenario. The "binary magic-byte" check from #119 still applies for symlink-target inspection; this fix is specifically for the importable-Python-package branch.
